lib/web/integrations_samlidp.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/integrations/samlidp/samlidpconfig"
+	libutils "github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+func (h *Handler) gcpWorkforceConfigScript(w http.ResponseWriter, r *http.Request, p httprouter.Params) (any, error) {
+	queryParams := r.URL.Query()
+
+	samlIdPMetadataURL := fmt.Sprintf("https://%s/enterprise/saml-idp/metadata", h.PublicProxyAddr())
+	// validate queryParams params
+	if err := (samlidpconfig.GCPWorkforceAPIParams{
+		OrganizationID:     queryParams.Get("orgId"),
+		PoolName:           queryParams.Get("poolName"),
+		PoolProviderName:   queryParams.Get("poolProviderName"),
+		SAMLIdPMetadataURL: samlIdPMetadataURL,
+	}).CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// The script must execute the following command:
+	// teleport integration configure samlidp gcp-workforce
+	argsList := []string{
+		"integration", "configure", "samlidp", "gcp-workforce",
+		fmt.Sprintf("--org-id=%s", libutils.UnixShellQuote(queryParams.Get("orgId"))),
+		fmt.Sprintf("--pool-name=%s", libutils.UnixShellQuote(queryParams.Get("poolName"))),
+		fmt.Sprintf("--pool-provider-name=%s", libutils.UnixShellQuote(queryParams.Get("poolProviderName"))),
+		fmt.Sprintf("--idp-metadata-url=%s", libutils.UnixShellQuote(samlIdPMetadataURL)),
+	}
+	script, err := oneoff.BuildScript(oneoff.OneOffScriptParams{
+		TeleportArgs:   strings.Join(argsList, " "),
+		SuccessMessage: "Success! You can now go back to the browser to complete the database enrollment.",
```

Comment:
This should probably something other than "database enrollment".

Reply:
Uh Oh! updated in https://github.com/gravitational/teleport/pull/39714/commits/cb64da8b839292dda925a36b514c985ea1c9d5a9